src/ng/directive/input.js

```diff
@@ -1494,7 +1490,15 @@ function createDateInputType(type, regexp, parseDate, format) {
     }
 
     function parseObservedDateValue(val) {
-      return isDefined(val) && !isDate(val) ? parseDate(val) || undefined : val;
+      return isDefined(val) && !isDate(val) ? parseDateAndConvertTimeZoneToLocal(val) || undefined : val;
+    }
+
+    function parseDateAndConvertTimeZoneToLocal(value, previousDate) {
+      var parsedDate = previousDate ? parseDate(value, previousDate) : parseDate(value);
```

Comment:
Why not simply ` = parseDate(value, previousDate)`? If the previousDate is undefined, it's not used.

Reply:
They are same but they only different in case you check arguments
```javascript
function parse(a, b){
    console.log(arguments.length)
}

parse(1) // it will log 1
parse(1, undefined) //it will log 2
```
if someone in the future decide do something depend on the arguments.length it will not be the same.

Reply:
I doubt this will happen. Even more so as this is an internal API. Our tests should also catch any problems this might cause.